src/fsharp/NameResolution.fsi

```diff
@@ -571,10 +584,10 @@ type AfterResolution =
     | RecordResolution of Item option * (TyparInst -> unit) * (MethInfo * PropInfo option * TyparInst -> unit) * (unit -> unit)
 
 /// Resolve a long identifier occurring in an expression position.
-val internal ResolveLongIdentAsExprAndComputeRange  : TcResultsSink -> NameResolver -> range -> AccessorDomain -> NameResolutionEnv -> TypeNameResolutionInfo -> Ident list -> Item * range * Ident list * AfterResolution
+val internal ResolveLongIdentAsExprAndComputeRange  : TcResultsSink -> NameResolver -> range -> AccessorDomain -> NameResolutionEnv -> TypeNameResolutionInfo -> Ident list -> Item * range * Ident list * TypeInst * AfterResolution
 
 /// Resolve a long identifier occurring in an expression position, qualified by a type.
-val internal ResolveExprDotLongIdentAndComputeRange : TcResultsSink -> NameResolver -> range -> AccessorDomain -> NameResolutionEnv -> TType -> Ident list -> FindMemberFlag -> bool -> Item * range * Ident list * AfterResolution
+val internal ResolveExprDotLongIdentAndComputeRange : TcResultsSink -> NameResolver -> range -> AccessorDomain -> NameResolutionEnv -> TType -> Ident list -> TypeNameResolutionInfo -> FindMemberFlag -> bool -> Item * range * Ident list * TypeInst * AfterResolution
```

Comment:
1. `ResolveLongIdentAsExprAndComputeRange` 
2. `ResolveExprDotLongIdentAndComputeRange` 

They have really similar names and are confusing when seeing their use in type checker; I always think of the wrong one.

The first one is to resolve an identifier, which could contain dots, as an expression: `System.Diagnostics.Stopwatch()`
The second is similar, but it's intent is to resolve an identifier that is off an expression: `stopwatch.` -> this is usually called after `Dot` in syntax, hence the name `ResolveExpr*Dot*LongIdentAndComputeRange` - However, I think `Dot` should be removed and maybe the names should be:
1. `ResolveExprLongIdent` (name already exists so maybe find a way to unify?) - no more `As` or `ComputeRange`
2. `ResolveExprLongIdentFromExpr` - no more `Dot` or `ComputeRange`

Reply:
It probably wouldn't be hard. My thinking is that unifying it could come with a performance cost, but I haven't proven it yet. I'll look at this a bit more.